pkg/controller/workload/job/job_controller.go

```diff
@@ -184,7 +196,7 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 
 	// 4. Handle a not finished job
```

Comment:
FYI, if a Job finishes fast, we might never add the `PodsReady` condition. Which is fine, but we need to consider that in the rest of the implementation.